### PR TITLE
feat: add ClassificationReport metric for binary, multiclass, and multilabel tasks

### PR DESCRIPTION
**Pull request by jlaportebot**
## Summary

Adds a `ClassificationReport` metric to torchmetrics, providing sklearn-style classification reports with precision, recall, F1-score, and support for each class, plus micro/macro/weighted averages.

### New classes
- `BinaryClassificationReport` — binary classification reports
- `MulticlassClassificationReport` — multiclass classification reports  
- `MultilabelClassificationReport` — multilabel classification reports
- `ClassificationReport` — task-agnostic wrapper (like `Accuracy`, `F1Score`)

### New functional API
- `binary_classification_report()`
- `multiclass_classification_report()`
- `multilabel_classification_report()`

### Features
- **output_format**: `"dict"` (nested dict) or `"flat"` (single-level dict with prefixed keys)
- **target_type**: `"auto"`, `"binary"`, `"multiclass"`, `"multilabel"` for the wrapper class
- **Multidim support**: handles (N, C, D) inputs with `multidim_average`
- **ignore_index**: exclude specific class indices from computation
- **Samplewise**: per-sample reports via `average="samplewise"`
- Per-class precision, recall, f1_score, support
- Micro, macro, weighted averages (binary also includes binary-specific average)

### Tests
153 tests covering:
- All task types (binary, multiclass, multilabel)
- All input formats (labels, probs, logits)
- Single-dim and multi-dim inputs
- Dict and flat output formats
- Incremental state updates
- Zero-division edge cases
- ignore_index
- Samplewise output
- Functional vs class parity
- Validated against sklearn.metrics.classification_report

### Files changed
- `src/torchmetrics/classification/classification_report.py` (615 lines) — class-based metrics
- `src/torchmetrics/functional/classification/classification_report.py` (397 lines) — functional API
- `tests/unittests/classification/test_classification_report.py` (914 lines) — comprehensive tests
- `__init__.py` updates for proper exports

All 153 tests pass locally. Ruff lint clean.

🤖 This PR was generated with AI assistance. The code, tests, and documentation were written by an automated coding agent.

### CODE REVIEW

File: src/torchmetrics/functional/classification/classification_report.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+    for c in range(num_classes):
+        result[str(c)] = {
+            "precision": per_class["precision"][c],
+            "recall": per_class["recall"][c],
+            "f1_score": per_class["f1_score"][c],
+            "support": per_class["support"][c],
+        }
+
+    # Summary averages (always include all)
+    result["micro"] = _compute_average(per_class, "micro", tp, fp, fn)
```
Comment (justusschock) [inline src/torchmetrics/functional/classification/classification_report.py:306]: Not sure if we always want to include all of them. As you also wouldn't include all of them when calculating these individually. Shouldn't be much calculation overhead to call this function once per averaging

File: src/torchmetrics/classification/classification_report.py [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+if not _MATPLOTLIB_AVAILABLE:
+    __doctest_skip__ = [
+        "BinaryClassificationReport.plot",
+        "MulticlassClassificationReport.plot",
+        "MultilabelClassificationReport.plot",
+        "ClassificationReport.plot",
+    ]
+
+
+class BinaryClassificationReport(BinaryStatScores):
```
Comment (justusschock) [inline src/torchmetrics/classification/classification_report.py:39]: Can we base these on a metric collection? This would enable automatic sharing of intermediate representations and be a bit cleaner with the dict output (e.g. it would enable typechecks against metriccollection on the consumer side)

**Review by justusschock — COMMENTED** [review]
Love the idea. Few comments
